joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

joint_x_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - joint_x
  gains:
    joint_x: {p: 1000.0, i: 500.0, d: 10.0}
  constraints:
    goal_time: 0.0
    stopped_velocity_tolerance: 0.02
    joint_x:
      trajectory: 0.05
      goal: 0.02

joint_y_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - joint_y
  gains:
    joint_y: {p: 1000.0, i: 500.0, d: 10.0}
  constraints:
    goal_time: 0.0
    stopped_velocity_tolerance: 0.02
    joint_y:
      trajectory: 0.05
      goal: 0.02

joint_z_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - joint_z
  gains:
    joint_z: {p: 5000.0, i: 2000.0, d: 100.0}
  constraints:
    goal_time: 0.0
    stopped_velocity_tolerance: 0.05
    joint_z:
      trajectory: 0.1
      goal: 0.05
